### DISCUSSION TITLE
Q on translation between (x, y) and (lat, lon) coordinates

### DISCUSSION THREAD

**Discussion by yaofuzhou** [domlysz/BlenderGIS#506] [Q&A] [answered]
May I kindly ask how are the (x, y) coordinates relate to the (lat, lon) coordinates? I understand that the 3D models supplied by Blender GIS has a known (lat, lon) coordinate for the origin, and the points of the 3D model is expressed in the (x, y, z) coordinates. How may I translate them to the (lat, lon) coordinates? In case approximations are involved. My area of interest has the dimension of up to 200 km.

Appreciate any help that may improve my understanding of this issue!

**Comment by domlysz** [comment, marked as answer]
Well that's what we call map projection
https://en.wikipedia.org/wiki/Map_projection
https://github.com/domlysz/BlenderGIS/wiki/Georeferencing-management

**Reply by yaofuzhou** [reply]
Sorry for a follow-up question -

Is it possible, when exporting the 3D model (I use the .stl format, for example), for the coordinates to use (lat, lon) rather than (x, y)?

Should I apply the inverse Mercator projection to the (x, y) coordinates myself; or, since the terrain and skyline data (probably?) come in (lat, lon), would I be able to access and/or export in terms of (lat, lon)?

Thank you @domlysz !

**Reply by baswein** [reply]
@yaofuzhou - just guessing here but my strategy would be to export your geometries back out as a shapefile from Blender and then use a GIS program such as QGIS to view the coordinates in lat long using the projection of your choice.

**Reply by yaofuzhou** [reply]
Thank you @baswein for your response! It has been a learning experience as I have not worked with Blender, GIS or any 3D related project at all.

My current understanding is that reprojection can be very computing-intensive, and they could lead to rounding issues if applied back and forth as in my old approach. Since my intention is to work with (lat, lon) coordinates, I should try to get the data directly from Open Street Maps or so rather than from reprojection from data exported from Blender. I can always get my data processing and analysis done outside Blender and than use Blender just for the rendering...

Forgive me for the rumbling but please feel free to correct me if I have made an obvious mistake.

Cheers!

**Reply by baswein** [reply]
With a GIS program re-projection is not too intensive in fact QGIS does it on the fly without you even noticing anything. QGIS has great tools to import and process OSM data. I have heard that there are rounding issues with blender and large distances but really am not familiar with that. What exactly are you trying to do?

**Reply by yaofuzhou** [reply]
Thank you again for telling me about QGIS @baswein.

I have been using pyproj to do the reproduction, which was somehow the slowest part of my program. My task, to begin with, is to get the terrain/skyline of a city of ~100km across in a customized format with vertices in (lat, lon, z) coordinates (EPSG:4326). The (naive) way I am doing it is:

1. Use BlenderGIS to import a manageable portion of the city.
2. Export to ASCII .stl format, in which vertices are in (x, y, z) coordinates (EPSG:3857).
3. Use Python with pyproj to reproject EPSG:3857 to EPSG:4326.
4. Repeat above for the next portion.
5. Rounding issue arises when I try to put adjacent regions with slight overlaps together. I have limited the number of decimal places in the (lat, lon) coordinates and have been able to remove duplicate data in the overlap.
6. In addition, as mentioned, pyproj takes over a day for the reprojections…

I deeply appreciate you telling me that this can be done in a much more efficient way!

**Reply by baswein** [reply]
Unfortunately I am probably as naive as you but I don't think the export to shapefile and QGIS will really work for you because you would still need to figure out how to get the shapefile back to stl.  Have you looked at BlenderOSM?

**Reply by yaofuzhou** [reply]
Not yet @baswein Sounds like another item on my to-know-about list.

In case it helps, the only reason I was working with ASCII .stl is that I can humanly read the vertices, and I can easily extract the (x,y,z) coordinates. This is no longer necessary.

In any case, the seemingly promising approach for me is to get terrain/skyline directly in the (lat, lon) format from OSM or so. Then other than the final rendering I can simply stick to (lat, lon).

Appreciate your time and hints!

**Reply by baswein** [reply]
If you want to send me an email - bas  at rdgland  com - we can continue this conversation by email in case we are bugging domlysz
